### DISCUSSION TITLE
How to get informations about a video within the code

### DISCUSSION THREAD

**Discussion by Sp3rick** [FreeTubeApp/FreeTube#5487] [Q&A]
Hi, im trying to implement a function to allow to import playlist from google exported csv files, the problem is that them contains only the videoId, and adding a playlist only using that will create a bunch of black videos with no title, no author recerence etc.. only the thumbnail that's dynamically loaded
To fix that i could fetch data of video before getting added to the playlist using the api but i can't figure out what function use to accomplish that, i want to count the fact that user may ave different API Backend preferences so i cant use directly getVideoInformationInvidious

**Comment by absidue** [comment]
The main problem you'll have is that you'll only be able to do it for a small number of videos before getting blocked by YouTube. Which is why it hasn't been implemented yet.

**Reply by Sp3rick** [reply]
That can be implemented with a delay between a fetch and another, this is going to take longer to import big playlists but it's still a solution

**Comment by Sp3rick** [comment]
Im creating on src/render/helpers/api a file preferences.js that allow you to gather info from api without need to worry what preferences are and what api to use and what to do in case of fallback, preferences.js will handle that for you
